presto-main/src/main/java/com/facebook/presto/metadata/FunctionListBuilder.java

```diff
@@ -185,10 +184,9 @@ private boolean processScalarFunction(Method method)
         }
         SqlType returnTypeAnnotation = method.getAnnotation(SqlType.class);
         checkArgument(returnTypeAnnotation != null, "Method %s return type does not have a @SqlType annotation", method);
-        Type returnType = type(typeManager, returnTypeAnnotation);
-        Signature signature = new Signature(name.toLowerCase(ENGLISH), returnType.getTypeSignature(), Lists.transform(parameterTypes(typeManager, method), Type::getTypeSignature));
+        Signature signature = new Signature(name.toLowerCase(ENGLISH), parseTypeSignature(returnTypeAnnotation.value()), parameterTypeSignatures(method));
```

Comment:
processScalarFunction add FunctionInfo to registry. FunctionInfo should is bounded (isUnbound() returns false). However, having any literals makes the function unbounded.